### PR TITLE
task-5: moved to custom cartridge

### PR DESCRIPTION
**Pull request by Vladzio26**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (Vladzio26) [inline .vscode/launch.json:1]: Should be added to gitignore

File: storefront-reference-architecture-master/cartridges/app_practical_tasks/cartridge/templates/resources/cart.properties [hunk 2]
```diff
@@ -1,73 +1 @@
...
-title.your.shopping.cart=Your shopping cart
-
-msg.bundle.includes=Includes:
-
-error.invalid.coupon=Your shopping cart contains a coupon that is invalid at this time. Please remove the coupon before proceeding to checkout.
-error.invalid.tax=An error has occurred calculating taxes for one or more products in your cart.
-error.cart.or.checkout.error=Your shopping cart cannot be ordered since one or more of the products\r\n in your cart are not available in the requested quantity. Please check availability for each product.
-error.cart.expired=Your shopping cart has expired.
-error.card.invalid.shipments=Your order cannot be completed because at least one of the addresses for products is invalid.
+label.warning=Your cart total exceeds ${0}!
```
Comment (oleksandra-davydchuk) [inline storefront-reference-architecture-master/cartridges/app_practical_tasks/cartridge/templates/resources/cart.properties:1]: $ sign is not needed, just {0}

File: storefront-reference-architecture-master/cartridges/app_practical_tasks/cartridge/controllers/Cart.js [hunk 3]
```diff
@@ -1,17 +1,963 @@
...
 var server = require('server');
-var pageMetaData = require('*/cartridge/scripts/middleware/pageMetaData');
-var page = module.superModule;
-server.extend(page);
 
-server.append('Show', function (req, res, next) {
+var csrfProtection = require('*/cartridge/scripts/middleware/csrf');
+var consentTracking = require('*/cartridge/scripts/middleware/consentTracking');
+
+server.get('MiniCart', server.middleware.include, function (req, res, next) {
```
Comment (oleksandra-davydchuk) [inline storefront-reference-architecture-master/cartridges/app_practical_tasks/cartridge/controllers/Cart.js:8]: You don't need to copy all from controller file, in SFRA it should be only supplemented with your code

File: storefront-reference-architecture-master/cartridges/app_practical_tasks/cartridge/controllers/Cart.js [hunk 4]
```diff
@@ -1,17 +1,963 @@
...
+
+        if (currentBasket && currentBasket.allLineItems.length) {
+            reportingURLs = reportingUrlsHelper.getBasketOpenReportingURLs(currentBasket);
+        }
+
+        res.setViewData({ reportingURLs: reportingURLs });
+
+        var basketModel = new CartModel(currentBasket);
+
+        var maxBasketAmt = Site.current.getCustomPreferenceValue('max_basket_amount');
```
Comment (oleksandra-davydchuk) [inline storefront-reference-architecture-master/cartridges/app_practical_tasks/cartridge/controllers/Cart.js:210]: Where is metadata file with "max_basket_amount" preference?
It should be added in .xml file with usage description, default value, etc., to be imported to other sandboxes

Also it's better to start preference name with group name, for example group name is MY_TEST_GROUP
preference name is MY_TEST_GROUP_Max_basket_amount 
to recognize it when you have similar preferences for different features (like payment methods, etc.)

File: storefront-reference-architecture-master/cartridges/app_practical_tasks/cartridge/controllers/Cart.js (hunk 3 above)
Comment (oleksandra-davydchuk) [reply storefront-reference-architecture-master/cartridges/app_practical_tasks/cartridge/controllers/Cart.js:8]: Sorry, I see this changes in other PR

**Pull request merged**